silo/forms.py

```diff
@@ -57,6 +60,12 @@ class Meta:
         fields = ['id', 'name', 'description', 'tags', 'shared',
                   'share_with_organization', 'owner', 'workflowlevel1']
 
+    def clean_shared(self):
+        data = self.data.getlist('shared')
+        owner = self.data.get('owner')
+        if data and owner and owner in data:
+            raise ValidationError('You can not share table with owner.')
+        return data
```

Comment:
Always return a value to use as the new cleaned data, even if this method didn't change it.